### PR TITLE
bidirectional gRPC outputs - client update

### PR DESCRIPTION
**Pull request by fntlnz**
Co-Authored-By: Leonardo Di Donato <leodidonato@gmail.com>
Signed-off-by: Lorenzo Fontana <lo@linux.com>

**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area client
/area examples

**What this PR does / why we need it**:

Update this client to reflect the changes going into the server at via https://github.com/falcosecurity/falco/pull/1241 - 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
BREAKING CHANGE: the output service named `output.subscribe` is now split into two different methods, the service name is now `outputs`.
BREAKING CHANGE: `outputs.get`: open a connection and send a request, receive all the available events until now, and stop.
BREAKING CHANGE: `outputs.sub`: open a connection to stream many requests, for each one of them it streams back all the available events and keeps the stream and connection open.
```

### CODE REVIEW

**Comment by fntlnz** [comment]
Keeping the wip since there are still some changes we need to make and tests against the new server.

**Review by leodido — DISMISSED** [review]

File: examples/output_bidi/main.go
Comment (leodido) [inline examples/output_bidi/main.go]: Probably this should be reverted.

File: examples/output_bidi/main.go
Comment (leodido) [inline examples/output_bidi/main.go]: This is **WIP**

Comment (leogr) [reply examples/output_bidi/main.go]: Could you add more context here pls?

File: Makefile
Comment (fntlnz) [inline Makefile]: todo: change this back to master before merging this once the corresponding PR gets merged into Falco https://github.com/falcosecurity/falco/pull/1241

File: pkg/client/client.go
Comment (leogr) [inline pkg/client/client.go]:
```suggestion
				errCh <- fmt.Errorf("error closing stream other than EOF: %v", err)
```

**Review by leogr — COMMENTED** [review]
SGTM, just a minor suggestion to improve the error message

File: pkg/client/client.go
Comment (leogr) [reply pkg/client/client.go]: Solved by https://github.com/falcosecurity/client-go/pull/45/commits/59af564fe1774a101ab45c4ab9c600424885c0f9

**Review by leogr — DISMISSED** [review]
LGTM

**Comment by leodido** [comment]
/hold

waiting for https://github.com/falcosecurity/falco/pull/1241 to be merged in.

**Review by leodido — DISMISSED** [review]

**Review by leodido — DISMISSED** [review]

**Review by leogr — DISMISSED** [review]
:+1:

File: examples/output/main.go
Comment (bgarber) [inline examples/output/main.go:33]: This breaks the example, because it won't keep the connection open. The `fcs.Recv()` below relies in this connection open, so it will only get the events kept in memory by the Falco daemon and exiting in the next iteration of the `for` loop (because the connection is now closed).

Comment (leodido) [reply examples/output/main.go:33]: Hello @bgarber this is the intended use case of the `falco.outputs.service/get` RPC.

The connection gets opened, then all the Falco outputs in the stream are received until there are no more events, and finally, the connection gets closed.

File: Makefile
Comment (leodido) [reply Makefile]: Done, about to commit!

File: examples/output/main.go
Comment (fntlnz) [reply examples/output/main.go:33]: @bgarber thanks for checking this out! Yes we changed the way outputs works by splitting into two different rpc calls:

- `get` connects, gets all the events available and then closes
- `sub` is the same as the old `Subscribe`, connects, gets all the events and then keeps getting them until something stops the process

If you want to see an example with the old behavior look at:
- `examples/output_unix_socket_bidi`
- `examples/output_bidi`

**Review by leodido — APPROVED** [review]
🌮

File: examples/output/main.go
Comment (bgarber) [reply examples/output/main.go:33]: @fntlnz Thank you very much for clarifying this for me!

Comment (leodido) [reply examples/output/main.go:33]: Thanks to you too @bgarber

**Pull request merged**
